### DISCUSSION TITLE
Framework usage examples (please add suggestions)

### DISCUSSION THREAD

**Discussion by focustense** [focustense/StardewUI#10] [General]
I'd like to provide a good, clear set of maybe 5-10 examples covering everything from introductory material (_"how do I create a simple menu"_) to very advanced usage (nested repeaters, included views, and generally anything with complex behavior). The best examples would be UIs that modders have actually made and found somewhat difficult, or are thinking of making but haven't pulled the trigger - again, because difficult.

Please add any suggestions here. There is no need to actually create the examples - I will do that as part of the docs/testing - I'm simply looking to understand what people might want to do with the framework.

Note - I'll probably use my own [Garbage In Garbage Can](https://github.com/focustense/StardewAutoTrash) as one of the examples - redone as StarML, of course - and see that as kind of a middling example; it's not terribly complex but it does have a grid, repeater, scrolling, etc., so it's not beginner-level either. Suggestions on both the easier and harder end of the spectrum would be great.

P.S. This doesn't have to be limited to menus. [AFS](https://github.com/focustense/StardewFishingSea) uses StardewUI for some of the HUD as well. Right now, there isn't an API for creating a HUD through StarML, but it's easy to add if there are any compelling examples. Examples could also include config menus that were difficult or involved tradeoffs in GMCM; AFS has its own custom config and the [Radial Menu](https://github.com/focustense/StardewRadialMenu) config also involves a bunch of nutty hacks to GMCM that would be less of a mess in StardewUI. I'm happy to take on any examples _like_ these from other people's mods or idea bins.

**Comment by Mushymato** [comment]
- Form/Config menu, with the usual text input, checkbox, dropdowns, keybind, buttons 
- Grid view of all fruit objects in the game, demonstrate data binding 
- Tabbed menu, akin to the vanilla pause menu.
- Overlay/Modal as child menus
- HUD elements

**Reply by focustense** [reply]
Thanks for this. Just for clarification, this entire discussion is about what you're referring to as data binding (that is the "framework", as opposed to the "library" or "shared project" which we've been using up 'til now). While the library/shproj is going to continue to remain available as a submodule, at least for the foreseeable future, I'm expecting that 9 out of 10 users will have a much easier time with the framework (same as using React/Angular/etc. vs. building a whole app in vanilla JavaScript, or building a WPF app using MVVM vs. using all the control types directly with code behind; you can do the latter, but it's not a ton of fun.)

Thanks again!

**Comment by OWL4C** [comment]
Please add a Minimal Reproducible Example MRE too, just a complete mod with nothing else than a simple UI Popup when a Game Event (eg Keypress) happens. I have spent multiple hours now looking at the TestMod without getting a single Working UI In Game, it is just way too convoluted and overloaded to understand when you are already have no grasp.

**Reply by focustense** [reply]
I'm unclear as to what would have taken several hours? There's a full step-by-step [getting started](https://focustense.github.io/StardewUI/getting-started/) guide in the docs and the [examples page](https://focustense.github.io/StardewUI/examples/) has the condensed code and markup for 4 out of the 6 examples.

The `TestMod` isn't meant to be a copy-and-paste example. It's the container mod for all the individual examples, and a general sandbox for staging new features.

**Reply by OWL4C** [reply]
Either i skipped reading multiple relevant steps for multiple times now or something obvious isn't obvious for me :(, see my other Q&A
